security-setup

```diff
@@ -297,11 +304,14 @@ class Consul(Component):
     def gossip_key(self):
         "gossip key"
         with self.modify_security() as config:
-            if 'consul_gossip_key' not in config:
-                config['consul_gossip_key'] = self.random(16)
-                print('set gossip key')
+            if not self.args.no_ssl:
+                if 'consul_gossip_key' not in config:
+                    config['consul_gossip_key'] = self.random(16)
+                    print('set gossip key')
+                else:
+                    print('gossip key already set')
             else:
-                print('gossip key already set')
+                print('disable ssl security')
```

Comment:
The Consul gossip key doesn't have anything to do with SSL - why is it being modified as part of this PR? If it really does need to modified, that role should be changed to remove the gossip key if `enable_ssl` is not set as well.